hack/clean.sh

```diff
@@ -29,6 +29,6 @@ for source_tree in $@; do
   grep -lr '// Code generated by lister-gen. DO NOT EDIT' "$(dirname "$source_tree")" | xargs rm -f
 done
 
-if [ -d "$PWD/hack/api-reference" ]; then
-  find ./hack/api-reference/ -type f -name "*.md" -exec rm '{}' \;
+if [ -d "$PWD/docs/api-reference" ]; then
+  find ./docs/api-reference/ -type f -name "*.md" -exec rm '{}' \;
```

Comment:
This change will also be relevant for extensions when revendoring this PR. Can you add a respective release note for developers, please?